docker/CMSRucioClient/scripts/updateRSEs.py

```diff
@@ -0,0 +1,180 @@
+#! /bin/env python
+# Script for managing (creating/updating/deleting) the RSEs and their attributes 
+# Initially will use PhEDEx nodes information as input,
+# should be easily transformed/extended for using other sources. 
+
+import argparse
+import logging
+
+import requests
+import json
+import re
+
+from functools import wraps
+
+from rucio.client.accountclient import AccountClient
+from rucio.client.rseclient import RSEClient
+from rucio.common.exception import Duplicate, RSEProtocolPriorityError, \
+	RSEProtocolNotSupported, RSENotFound, InvalidObject
+
+DATASVC_URL='http://cmsweb.cern.ch/phedex/datasvc/json/prod/' 
+prog = re.compile('.* -service (.*?) .*')
+
+def setup_logger(logger):
+	""" Code borrowed from bin/rucio-admin
+	"""
+	logger.setLevel(logging.DEBUG)
+	hdlr = logging.StreamHandler()
+	def emit_decorator(fcn):
+		def func(*args):
+			if True:
+				formatter = logging.Formatter("%(message)s")
+			else:
+				levelno = args[0].levelno
+				if levelno >= logging.CRITICAL:
+					color = '\033[31;1m'
+				elif levelno >= logging.ERROR:
+					color = '\033[31;1m'
+				elif levelno >= logging.WARNING:
+					color = '\033[33;1m'
+				elif levelno >= logging.INFO:
+					color = '\033[32;1m'
+				elif levelno >= logging.DEBUG:
+					color = '\033[36;1m'
+				else:
+					color = '\033[0m'
+				formatter = logging.Formatter('{0}%(asctime)s %(levelname)s [%(message)s]\033[0m'.format(color))
+			hdlr.setFormatter(formatter)
+			return fcn(*args)
+		return func
+	hdlr.emit = emit_decorator(hdlr.emit)
+	logger.addHandler(hdlr)
+
+def exception_handler(function):
+	"""Code borrowed from bin/rucio-admin
+	"""
+	@wraps(function)
+	def new_funct(*args, **kwargs):
+		try:
+			return function(*args, **kwargs)
+		except InvalidObject as error:
+			logger.error(error)
+			return error.error_code
+	return new_funct
+
+# Functions for getting PhEDEx information: 
+
+def PhEDEx_node_to_RSE(node, test_tag = '111debug'):
+	""" Translates PhEDEx node names to RSE names. 
+	Make sure new names comply with the policies defined in: 
+	./lib/rucio/common/schema/cms.py
+	./lib/rucio/core/permission/cms.py
+	Because once created RSE name can't be reused, allow to postpend
```

Comment:
Why do we want to append the string?

I assume we'll have to continuously update and sync, just like ATLAS does -- we can always do an update for pre-existing RSEs instead of recreating them.

Reply:
You're right, and the name suggests the constant updates. 
The added suffix for testing add/delete . I found no way to resurrect the RSE (via client) once it was deleted as the name could not be reused. 
This could also be useful if we want to introduce more namespaces for different purposes, like ATLAS does (USERDATA, SCRATCHDATA, etc).